### PR TITLE
[FIX] Use older vkd3d when vulkan 1.1 is detected

### PR DESCRIPTION
**Pull request by Etaash-mathamsetty**
title

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by Etaash-mathamsetty** [comment]
i think we might have to upload a 2.6 binary to the heroic fork, because the one in the official fork is a .tar.zst or something

File: src/backend/tools.ts [hunk 1]
```diff
@@ -529,3 +529,39 @@ function getDxvkUrl(): string {
...
+ */
+function getVkd3dUrl(): string {
+  if (!isLinux) {
+    return ''
+  }
+
+  if (any_gpu_supports_version([1, 3, 0])) {
+    const instance_version = get_vulkan_instance_version()
+    if (instance_version && semverLt(instance_version.join('.'), '1.3.0')) {
+      // FIXME: How does the instance version matter? Even with 1.2, newer DXVK seems to work fine
```
Comment (CommandMC) [inline src/backend/tools.ts:545]:
```suggestion
      // FIXME: How does the instance version matter? Even with 1.2, newer VKD3D seems to work fine
```
:^)

File: src/backend/utils/graphics/vulkan/index.ts [hunk 2]
```diff
@@ -1,22 +1,26 @@
 import { vulkanHelperBin } from 'backend/constants'
-import { spawnSync } from 'child_process'
+import { SpawnSyncReturns, spawnSync } from 'child_process'
 import { gte as semverGte } from 'semver'
 
 type VulkanVersion = [maj: number, min: number, patch: number]
 
+let instance_version: SpawnSyncReturns<string> | null = null
+let physical_versions: SpawnSyncReturns<string> | null = null
```
Comment (CommandMC) [inline src/backend/utils/graphics/vulkan/index.ts:8]: Could use something like
```suggestion
let instance_version: ReturnType<typeof spawnSync> | null = null
let physical_versions: ReturnType<typeof spawnSync> | null = null
```
here to always have the correct type & avoid the import

**Comment by CommandMC** [comment]
Not sure if we can release a 2.6 version in our repo, as that'll make *that* release the latest one (unless GitHub compares SemVer versions to determine which release is the latest, which AFAIK it doesn't do)

**Comment by Etaash-mathamsetty** [comment]
> Not sure if we can release a 2.6 version in our repo, as that'll make _that_ release the latest one (unless GitHub compares SemVer versions to determine which release is the latest, which AFAIK it doesn't do)

release 2.6 and then release 2.9 afterward lol

**Comment by imLinguin** [comment]
@CommandMC I'm pretty sure we can release it, when creating a release we just need to make sure we disable `set as latest release`. Then GitHub will use semantic versioning to determine latest one

**Review by imLinguin — APPROVED** [review]

**Review by CommandMC — APPROVED** [review]

**Pull request merged**